enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,702 @@
+   credentials. -->
+   
+This section outlines the workflow for enforcing the proposed scheduling policies that prevent non-control plane workloads from being scheduled on control plane nodes and allow for exceptions based on administrative configurations.
+
+#### Actors
+
+- Cluster Administrator: Responsible for configuring scheduling policies, managing exceptions, and overseeing cluster operations.
+- User: Individuals or services attempting to deploy workloads within the cluster.
+- Admission Controller: The mechanism that intercepts pod scheduling requests to enforce scheduling policies.
+
+#### Workflow Steps
+
+1. Policy Configuration:
+
+- The cluster administrator defines and configures the scheduling policies using a new or extended configuration resource within the OpenShift API. 
+- The admission plugin configuration will be stored under KAO [configuration](https://github.com/openshift/cluster-kube-apiserver-operator/blob/189ce0e7d47864366c741d3d5b9bd9c820421c14/bindata/assets/config/defaultconfig.yaml#L4C3-L4C15). The configuration will not be writable in the first iteration. There's going to be a predefined list of authorized SAs/users for all openshift components allowed to run on control plane nodes. Admins will be allowed to extend the authorized list of SAs/users only through config.openshift.io/v1/scheduler singleton object.
+- The policy configuration will be TechPreviewNoUpgrade feature gated. The feature gate name needs to be `OpenShift` prefixed.
+
+2. Pod Scheduling Request:
+
+- A user creates a new pod or updates an existing pod specification, including scheduling preferences such as node selectors or tolerations.
+- The request is submitted to the Kubernetes API server.
+
+3. Admission Control:
+
+- The admission controller intercepts the pod scheduling request before it is written to the etcd database.
+    - The controller evaluates the request against the configured scheduling policies. This includes checking:
+      - if the pod targets any control plane node and
+      - whether the pod's service account and requestor are on the allowlist.
+
+4. Policy Enforcement:
+
+- If the request violates the scheduling policies:
+  - The admission controller rejects the request.
+  - A meaningful error message is returned to the user, explaining the policy violation and suggesting corrective actions.
+- If the request complies with the scheduling policies or is exempt:
+  - The request is approved.
+  - The pod is scheduled according to its specified preferences and Kubernetes scheduling algorithms.
+
+5. Emergency Overrides (Optional):
+
+- In cases where temporary overrides of scheduling policies are necessary, the cluster administrator can apply a temporary configuration change.
+- The admission controller processes requests based on the updated policies until the override is removed.
+- Any actor with pre-granted permissions to perform an emergency override will be part of the list of authorized service accounts/users.
```

Comment:
From Jan comment: Running `oc debug modes/???` as an example. Only admins should be able to run it. Meaning, we need to allow certain (built-in?) users  (e.g. kube admin) to be always allows to assign pods to any node. The emergency overrides (e.g. extending the list of always allowed users) could be designed during another iteration of the implementation?

Reply:
> The emergency overrides (e.g. extending the list of always allowed users) could be designed during another iteration of the implementation?

We need to make sure the emergency case and ability to extend the list is well known common knowledge. 
We need to create the list of all must-be-in-the-default-list users before the first iteration is released. Including list of users coming from our layered products built on top of an OpenShift cluster.